lib/elixir/lib/stream.ex

```diff
@@ -1137,11 +1137,49 @@ defmodule Stream do
     lazy(enum, %{}, fn f1 -> R.uniq_by(fun, f1) end)
   end
 
+  @doc """
+  Builds a stream from an index, either starting from offset, or given by function.
+
+  May receive a function or an integer offset.
+
+  If an `offset` is given, it will emit elements from offset.
+
+  If a `function` is given, it will invoke the function with
+  elements from offset.
+
+  ## Examples
+
+      iex> Stream.from_index() |> Enum.take(3)
+      [0, 1, 2]
+
+      iex> Stream.from_index(1) |> Enum.take(3)
+      [1, 2, 3]
+
+      iex> Stream.from_index(fn x -> x * 10 end) |> Enum.take(3)
+      [0, 10, 20]
+  """
+  @doc since: "1.17.0"
```

Comment:
```suggestion
  @doc since: "1.17.0"
  @spec from_index(integer) :: Enumerable.t(integer)
  @spec from_index((integer -> return_value) :: Enumerable.t(return_value) when return_value: term
```